plugin/storage/es/spanstore/reader.go

```diff
@@ -668,14 +671,18 @@ func (s *SpanReader) buildOperationNameQuery(operationName string) elastic.Query
 
 func (s *SpanReader) buildTagQuery(k string, v string) elastic.Query {
 	objectTagListLen := len(objectTagFieldList)
-	queries := make([]elastic.Query, len(nestedTagFieldList)+objectTagListLen)
+	queries := make([]elastic.Query, objectTagListLen)
 	kd := s.spanConverter.ReplaceDot(k)
 	for i := range objectTagFieldList {
 		queries[i] = s.buildObjectQuery(objectTagFieldList[i], kd, v)
 	}
-	for i := range nestedTagFieldList {
-		queries[i+objectTagListLen] = s.buildNestedQuery(nestedTagFieldList[i], k, v)
+	if !s.allTagsAsFields {
+		for i := range nestedTagFieldList {
+			queries = append(queries, s.buildNestedQuery(nestedTagFieldList[i], k, v))
+		}
 	}
+	// Nested Logs Query
```

Comment:
explain why this is separate?

Reply:
So we had all the nested fields query together initally (tags , process tags , logs) . When we set AllTagsAsFields as true , we still want to query the nested logs field.

Reply:
is it because with AllTagsAsFields=true the tags from logs are still written as nested? Then how does this fix prevent the reported issue? When a span has no logs, it has no `tags:[]` in the document, but the query would still be looking for that?

Reply:
Yes, when AllTagsAsFields is true, logs are written as nested field. Arent logs and tags two independent fields in span context ? The change that I made does not search in the tags field and the process.tags field when allTagsAsFields is true. The reported issue said that they had a different mapping configuration than the jaeger default configuration and hence it never populated the tags field.

Reply:
that's my point - we should have a test that reproduces the reported issue in the first place (maybe as a new integration test), before we try fixing it. Otherwise how do you know that these changes are fixing it?

Reply:
Right, I understand that. Its however redundant to query the tags field when all tags as fields is true. I will work on the integration test!